repo/open.go

```diff
@@ -183,51 +186,125 @@ func openDirect(ctx context.Context, configFile string, lc *LocalConfig, passwor
 	return r, nil
 }
 
-// openWithConfig opens the repository with a given configuration, avoiding the need for a config file.
-// nolint:funlen,gocyclo
-func openWithConfig(ctx context.Context, st blob.Storage, lc *LocalConfig, password string, options *Options, caching *content.CachingOptions, configFile string) (DirectRepository, error) {
-	caching = caching.CloneOrDefault()
-
-	// Read repo blobs, potentially from cache.
-	fb, rb, err := readAndCacheRepositoryBlobs(ctx, st, caching.CacheDirectory, lc.FormatBlobCacheDuration)
+// nolint:gocritic
+func readAndCacheRepoConfig(
+	ctx context.Context,
+	st blob.Storage,
+	password string,
+	cacheOpts *content.CachingOptions,
+	validDuration time.Duration,
+) (
+	fb []byte,
+	cacheTime time.Time,
+	f *formatBlob,
+	repoConfig *repositoryObjectFormat,
+	formatEncryptionKey []byte,
+	err error,
+) {
+	// Read format blob, potentially from cache.
+	fb, cacheTime, err = readAndCacheRepositoryBlobBytes(ctx, st, cacheOpts.CacheDirectory, FormatBlobID, validDuration)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to read repository blobs")
+		return nil, time.Time{}, nil, nil, nil, errors.Wrap(err, "unable to read format blob")
 	}
 
-	if err = writeCacheMarker(caching.CacheDirectory); err != nil {
-		return nil, errors.Wrap(err, "unable to write cache directory marker")
+	if err = writeCacheMarker(cacheOpts.CacheDirectory); err != nil {
+		return nil, time.Time{}, nil, nil, nil, errors.Wrap(err, "unable to write cache directory marker")
 	}
 
-	f, err := parseFormatBlob(fb)
+	f, err = parseFormatBlob(fb)
 	if err != nil {
-		return nil, errors.Wrap(err, "can't parse format blob")
+		return nil, time.Time{}, nil, nil, nil, errors.Wrap(err, "can't parse format blob")
 	}
 
 	fb, err = addFormatBlobChecksumAndLength(fb)
 	if err != nil {
-		return nil, errors.Errorf("unable to add checksum")
+		return nil, time.Time{}, nil, nil, nil, errors.Errorf("unable to add checksum")
 	}
 
-	formatEncryptionKey, err := f.deriveFormatEncryptionKeyFromPassword(password)
+	formatEncryptionKey, err = f.deriveFormatEncryptionKeyFromPassword(password)
 	if err != nil {
-		return nil, err
+		return nil, time.Time{}, nil, nil, nil, err
 	}
 
-	repoConfig, err := f.decryptFormatBytes(formatEncryptionKey)
+	repoConfig, err = f.decryptFormatBytes(formatEncryptionKey)
 	if err != nil {
-		return nil, ErrInvalidPassword
+		return nil, time.Time{}, nil, nil, nil, ErrInvalidPassword
+	}
+
+	return fb, cacheTime, f, repoConfig, formatEncryptionKey, nil
+}
+
+// ReadAndCacheRepoConfig loads the config from cache and returns the cache
```

Comment:
does this need to be exported?

Reply:
not atm, maybe on the next PR. Will unnexport for now

Reply:
dropped this function, I needed it on the POC PR for some reason but not anymore